controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -179,6 +197,9 @@ func (r *RegisteredClusterReconciler) getManagedCluster(regCluster *singaporev1a
 		return managedClusterList.Items[0], nil
 	}
 
+	if regCluster.DeletionTimestamp != nil {
```

Comment:
Not sure about this check? It seems misleading that a deleted registered cluster would end up returning a "correct managed cluster not found" error message. Could you share some more detail on what this is for? Thanks!

Reply:
Added this check, if regcluster is deleted and in deletion process managedcluster is deleted. In this case it was returning error message correct managed cluster is not found

Reply:
Ah I had this backwards in my head. Thanks for clarifying!